docs/spec/lcd/api.md

```diff
@@ -0,0 +1,847 @@
+
+Parameters: null
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+    	"name": "test",
+          "address": "cosmosaccaddr1thlqhjqw78zvcy0ua4ldj9gnazqzavyw4eske2",
+          "pub_key": "cosmosaccpub1zcjduc3qyx6hlf825jcnj39adpkaxjer95q7yvy25yhfj3dmqy2ctev0rxmse9cuak"
+    },
+    "rest api": "2.0"
+}
+
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot find corresponding name",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /keys/{name} - PUT
+
+url: /keys/{name}, Method: PUT
+
+Functionality: Update key password
+
+Parameters: 
+
+| Parameter       | Type   | Default | Required | Description     |
+| --------------- | ------ | ------- | -------- | --------------- |
+| old_password    | string | null    | true     | password before |
+| new_password    | string | null    | true     | password before |
+| repeat_password | string | null    | true     | password before |
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "updated":name
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot update the corresponding key",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /keys/{name} - DELETE
+
+url: /keys/{name}, Method: DELETE
+
+Functionality: Delete key from keystore
+
+Parameters:
+
+| Parameter | Type   | Default | Required | Description      |
+| --------- | ------ | ------- | -------- | ---------------- |
+| password  | string | null    | true     | password of keys |
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     "deleted":name
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "cannot delete the corresponding key",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+## ICS20 - TokenAPI
+
+The TokenAPI exposes all functionality needed to query account balances and send transactions.
+
+### /balance/{account} - GET
+
+url: /balance/{account}, Method: GET
+
+Functionality:
+
+Parameters:
+
+* The above command returns JSON structured like this if success:
+
+```
+{
+    "error": "",
+    "code":200,
+    "result": {
+     {
+         "atom": 1000,
+         "photon": 500,
+         "ether": 20
+     }
+    },
+    "rest api": "2.0"
+}
+```
+
+* The above command returns JSON structured like this if fails:
+
+```
+{
+    "error": "Invalid account",
+    "code":500,
+    "result": {},
+    "rest api": "2.0"
+}
+```
+
+
+### /create_transfer - POST
+
+url: /create_transfer, Method: **POST**
+
+Functionality: transfer asset
+
+Parameters:
```

Comment:
Can we add a data byte field? Most token standards have it. It's useful to pass UserIds and such

Reply:
Also root of some trouble in Bitcoin ;)